exonum-java-binding/core/src/test/java/com/exonum/binding/core/runtime/ServiceNodeProxyIntegrationTest.java

```diff
@@ -74,6 +75,7 @@ void restrictSubmitTransaction() {
   }
 
   @Test
+  @Disabled("ECR-3828")
```

Comment:
Actually, I don't know why and what is the desired behaviour

Reply:
Because the function to which the transformation is applied is not invoked by the Node. If we could create a NodeProxy from java in tests, we would be able to write an integration test for that (with actual invocation). Until then, I'd just test:

```
  @Test
  void withSnapshot() {
    decorator.withBlockchainData(SNAPSHOT_FUNCTION);

    verify(node).withSnapshot(any(Function.class));
  }
```